### PR TITLE
MM-52833 : Migrate "components/admin_console/multiselect_settings.jsx" and tests to Typescript

### PR DESCRIPTION
**Pull request by Esterjudith**
…ize file

#### Summary

migrated components/admin_console/multiselect_settings.jsx to Typescript :
- Rename the files from jsx to tsx TypeScript extensions,  update the schema_admin_setting.jsx with the correct import. Did not see any associated test with this file. 

#### Ticket Link
fixes https://github.com/mattermost/mattermost-server/issues/23427
https://mattermost.atlassian.net/browse/MM-52833

#### Release Note

```release-note

```

### CODE REVIEW

File: webapp/package-lock.json [hunk 1]
```diff
@@ -1561,15 +1561,6 @@
         "eslint": ">=4.19.1"
       }
     },
-    "boards/node_modules/eslint-plugin-eslint-comments/node_modules/escape-string-regexp": {
```
Comment (M-ZubairAhmed) [inline webapp/package-lock.json:1564]: coud you please revert these changes

File: webapp/channels/src/components/admin_console/schema_admin_settings.jsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ import RequestButton from 'components/admin_console/request_button/request_butto
 import BooleanSetting from 'components/admin_console/boolean_setting';
 import TextSetting from 'components/admin_console/text_setting';
 import DropdownSetting from 'components/admin_console/dropdown_setting.jsx';
-import MultiSelectSetting from 'components/admin_console/multiselect_settings.jsx';
+import MultiSelectSetting from 'components/admin_console/multiselect_settings.tsx';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/schema_admin_settings.jsx:20]: you do not need to explicitly mention `tsx` here

File: webapp/channels/src/components/admin_console/multiselect_settings.tsx
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/multiselect_settings.tsx:1]: > In the Props interface I have the values data type as any[ ], I don't know what the data type for this should be. Where would I find this info?

Yes sometimes its hard to know what types a value may have. I usually run the app and add in few console.logs around the value to what values it might have.

> Would you please check if I have the correct syntax for the onChange in the Prop interface, also for line 37, and line 47, thank you.

Same suggestion as ⏫ try to see what values this function intakes as params

> I get the "Property 'error' does not exist on type 'Readonly<{}>" for this.state.error on line 80, thru my research on this error it looks like I would need to create another interface  interface State { error: string | null } and then add the interface to the component next to Props  export default class MultiSelectSetting extends React.PureComponent <Props, State>, is this correct?

Yes, you are correct check other ts files in the codebase with class components and you would see they also have `State` as type.

> I didn't see any test for this file, please let me know how you would like me to test.

If it doesnt have a test file. Its fine not to add one. This is applicable for this scenarios since we are not adding any few functionality in this PR but adding types and testing for types is done by ts compiler

> I didn't compile the tsx yet, I'm not familiar with microservices, so I'm not sure if I can do it directly in the webapp folder or in the root directory, please advice.

You don't have to compile but run the type checker. Go inside of webapp dir and run
```
npm run check-types
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/multiselect_settings.tsx
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/multiselect_settings.tsx:1]: Regarding the type of `values`, you could also try looking where that component is used. Despite the generic name, I think this component is only used for some language-related settings in the System Console

Comment (Esterjudith) [reply webapp/channels/src/components/admin_console/multiselect_settings.tsx:1]: Thank you so much M-Zubair and hmhealey these are great info.  I'll work on these things.

File: webapp/package-lock.json (hunk 1 above)
Comment (Esterjudith) [reply webapp/package-lock.json:1564]: Yes, no problem.

Comment (Esterjudith) [reply webapp/package-lock.json:1564]: I'm  running into an issue.  These updates happened because I ran npm install when I cloned the project, but if I don't do that then the web-server doesn't run because it "cannot find module 'chalk'", I though of running npm ci however this will delete the node_modules.  Do you have any recommendations? Thank you.

Comment (M-ZubairAhmed) [reply webapp/package-lock.json:1564]: yes you can revert these manually by copying the files from `master` branch

**Comment by M-ZubairAhmed** [comment]
Opened https://github.com/mattermost/mattermost-server/pull/23542 here
